src/tqec/plaquette/rpng_description.py

```diff
@@ -0,0 +1,245 @@
+        elif len(rpng_string) == 5:
+            r_str, a_str, p_str, n_str, g_str = tuple(rpng_string)
+        else:
+            raise ValueError('The rpng string must be exactly 4- or 5-character long.')
+        # Convert the characters into the enum attributes (or raise error).
+        r = ExtendedBasisEnum(r_str) if r_str in ExtendedBasisEnum._value2member_map_ else None
+        a = BasisEnum(a_str) if a_str in BasisEnum._value2member_map_ else None
+        p = BasisEnum(p_str) if p_str in BasisEnum._value2member_map_ else None
+        n = int(n_str) if n_str.isdigit() else None
+        g = ExtendedBasisEnum(g_str) if g_str in ExtendedBasisEnum._value2member_map_ else None
+        # Raise error if anythiong but '-' was used to indicate None.
+        if not r and r_str != '-':
+            raise ValueError('Unacceptable character for the R field.')
+        if not p and p_str != '-':
+            raise ValueError('Unacceptable character for the P field.')
+        if p and not a:
+            raise ValueError('Unacceptable ancilla basis in the extended format (i.e. rapng).')
+        if not n and n_str != '-':
+            raise ValueError('Unacceptable character for the N field.')
+        if not g and g_str != '-':
+            raise ValueError('Unacceptable character for the G field.')
+        return cls(r, p, n, g, a)
+
+    
+    def get_r_op(self) -> Optional[str]:
+        """Get the reset operation or Hadamard"""
+        op = self.r
+        if op == None:
+            return None
+        elif op.value in BasisEnum._value2member_map_:
+            return f'R{op.value.upper()}'
+        else:
+            return f'{op.value.upper()}'
+
+    def get_g_op(self) -> Optional[str]:
+        """Get the measurement operation or Hadamard"""
+        op = self.g
+        if op == None:
+            return None
+        elif op.value in BasisEnum._value2member_map_:
+            return f'M{op.value.upper()}'
+        else:
+            return f'{op.value.upper()}'
+
+
+@dataclass
+class RGN:
+    """Organize the prep and meas bases for the ancilla, together with the meas time
+    
+    The init time is assumed to be 0.
+    """
+    r: BasisEnum = BasisEnum.X
+    g: BasisEnum = BasisEnum.X
+    n: int = 6
+
+    @classmethod
+    def from_string(cls, rgn_string: str) -> 'RGN':
+        """Initialize the RGN object from a 3-character string"""
+        if len(rgn_string) != 3:
+            raise ValueError('The rgn string must be exactly 3-character long.')
+        r_str, g_str, n_str = tuple(rgn_string)
+
+        try:
+            r = BasisEnum(r_str)
+            g = BasisEnum(g_str)
+            n = int(n_str)
+            return cls(r, g, n)
+        except ValueError:
+            raise ValueError('Invalid rgn string.')
+
+
+@dataclass
+class RPNGDescription:
+    """Organize the description of a plaquette in RPNG format
+    
+    The corners of the square plaquette are listed following the order:
+    tl, tr, bl, br
```

Comment:
This should be explicitly written, and it might even be nice to have a visual representation. Something like
```suggestion
    top-left, top-right, bottom-left, bottom-right. That is a Z-shaped path on the plaquette data-qubits:
    
        +------------+
        | 1 -----> 2 |
        |        /   |
        |      /     |
        |    ∟       |
        | 3 -----> 4 |
        +------------+
```